tests/test_check_obsfiles_ert_yaml.py

```diff
@@ -0,0 +1,259 @@
+    df_production_data: pd.DataFrame =  pd.read_csv(_PRODUCTION_DATA_FILE_NAME,
+                                                    usecols=headers)
+    
+    df_production_data["date"] = pd.to_datetime(df_production_data["date"])
+    
+###############################################################################################                   
+###############################################################################################
+
+    #TODO Remormulate Schedules functions: I copied and pasted internal lines in _calculate_wconhist() and _calculate_wconinjh(). 
+    #     This functions requires well names and dates already asing by _calculate_compdat()
+    #     _calculate_welspecs()
+    #     
+    
+    schedule = Schedule(df_production_data = df_production_data)
+    for _, value in schedule._df_production_data.iterrows():
+        #start_date = schedule.get_well_start_date(value["WELL_NAME"])  
+        start_date = date(2005,10,1)
+        if value["TYPE"] == "WI" and start_date and value["date"] >= start_date:
+            schedule.append(
+                WCONINJH(
+                    date=value["date"],
+                    well_name=value["WELL_NAME"],
+                    inj_type="WATER",
+                    status=value["WSTAT"],
+                    rate=value["WWIR"],
+                    bhp=value["WBHP"],
+                    thp=value["WTHP"],
+                    inj_control_mode=schedule._inj_control_mode,
+                )
+            )
+        elif value["TYPE"] == "GI" and start_date and value["date"] >= start_date:
+            schedule.append(
+                WCONINJH(
+                    date=value["date"],
+                    well_name=value["WELL_NAME"],
+                    inj_type="GAS",
+                    status=value["WSTAT"],
+                    rate=value["WGIR"],
+                    bhp=value["WBHP"],
+                    thp=value["WTHP"],
+                    inj_control_mode=schedule._inj_control_mode,
+                )
+            )
+
+
+    #vfp_tables = self.get_vfp()
+    vfp_tables = {'K-3H': '1*', 'F-1H': '1*', 'F-2H': '1*'}
+    for _, value in schedule._df_production_data.iterrows():
+        #start_date = self.get_well_start_date(value["WELL_NAME"])
+        start_date = date(2005,10,1)
+        if value["TYPE"] == "OP" and start_date and value["date"] >= start_date:
+            schedule.append(
+                WCONHIST(
+                    date=value["date"],
+                    well_name=value["WELL_NAME"],
+                    status=value["WSTAT"],
+                    prod_control_mode=schedule._prod_control_mode,
+                    vfp_table=vfp_tables[value["WELL_NAME"]],
+                    oil_rate=value["WOPR"],
+                    water_rate=value["WWPR"],
+                    gas_rate=value["WGPR"],
+                    bhp=value["WBHP"],
+                    thp=value["WTHP"],
+                )
+            )
```

Comment:
Remormulate Schedules functions: I copied and pasted internal lines in _calculate_wconhist() and `_calculate_wconinjh()`.  This functions requires `well_names` and `start_dates` already asing by _calculate_compdat()     `_calculate_welspecs() `. So I include `well_names` and `start_dates` hard coded. But this is not the solution. We need to reformulate this part.